.circleci/config.yml

```diff
@@ -17,4 +17,4 @@ jobs:
       - run: mkdir build
```

Comment:
@NicholasCorrado When you have dependencies like gflags and glog that we download and build from source
they should be added to the circleci cache. This way they are built once on cicrleci and then saved for subsequent runs. For cmake and arrow this dropped our test run time from to 25 to 4 minutes.

All you need to do is add the paths of the built dependencies to the cache (like in lines 15 and 16). This commands saves the result of downloading and bulding.

When the circleci script is run again the cache is restored (line 8 and 9). After this restoration the hustle installation scripts are run again. The scripts have a conditions that checks if, here, arrow and cmake exists. If they do  downloading is skipped.
Please update the installation scipts of glog and gflag to work this way. We can have a chat to if you need more information.

Reply:
https://github.com/UWQuickstep/hustle/blob/d5af49a95e1bdad0d5e034e98cef6c6a6d74f275/third_party/install_third_party.sh#L27-L32

I added an `else` statement to skip downloading anything that already exists,

https://github.com/UWQuickstep/hustle/blob/d5af49a95e1bdad0d5e034e98cef6c6a6d74f275/third_party/install_third_party.sh#L58-L71

and I apply patches inside the loop so that we only attempt to patch if it exists. I also added the patch files, which I forgot to include earlier. 

CircleCi does not run install_third_party.sh, though. For now, we use the gflags and glog in this repository.

Reply:
Is there a reason not to run install_third_party.sh on circleci? if not you can add it on the circleci config

Reply:
The CMakeLists.txt files from the downloaded gflags and glog are not the same as the CMakeLists.txt files we currently have in this repository. For example, the downloaded glog actually depends on gflags, but that is not the case for the glog in this repository.

CMake will also complain unless we add this line to the gflags CMakeLists.txt:
https://github.com/UWQuickstep/hustle/blob/d5af49a95e1bdad0d5e034e98cef6c6a6d74f275/third_party/gflags/CMakeLists.txt#L11-L13

In brief, edits must be made to the cmake files to build hustle using gflags and glog downloaded using install_third_party.sh. I figured the edits would be straightforward, but it was having trouble getting gflags to link with the operator tests.

Reply:
Ok, I get it. The install_third_party.sh is not used right now to install glog and gflags. 
Then proceed as is and when the Cmake Issue is fixed we will update circleci too.